ui/src/components/common/SlidingSidebar/partials/EdgeDetails/index.tsx

```diff
@@ -55,13 +55,13 @@ export function EdgeDetails({ edgeId, watermarks }: EdgeDetailsProps) {
             )}
             {!!watermarks &&
               !!watermarks.length &&
-              watermarks.map((watermark: number, index: number) => (
+              watermarks.map((watermark: (string|number), index: number) => (
                 <TableRow key={index}>
                   <TableCell>{index}</TableCell>
                   <TableCell>
-                    {watermark < 0
+                    {Number(watermark) < 0
                       ? watermark
-                      : `${watermark} (${new Date(watermark).toISOString()})`}
+                      : `${watermark} (${new Date(Number(watermark)).toISOString()})`}
```

Comment:
Now the data is something like `{"data":[{"pipeline":"simple-pipeline","edge":"in-cat","watermarks":["1719990590574"],"isWatermarkEnabled":true,"from":"in","to":"cat"},{"pipeline":"simple-pipeline","edge":"cat-out","watermarks":["1719990589574"],"isWatermarkEnabled":true,"from":"cat","to":"out"}],"errMsg":null}`

The difference from original is the `watermarks`, which was `"watermarks":[1719990589574]`.

Verify if this works with original data format.

Reply:
It works perfectly with `restful` client.